improver/convection.py

```diff
@@ -350,3 +350,91 @@ def process(self, cube):
         )
 
         return output_cube
+
+
+class ConvectionRatioFromComponents(BasePlugin):
+    """
+    Diagnose the convective precipitation ratio by using differences between
+    convective and dynamic components.
+    """
+
+    def __init__(self):
+        self.convective = None
+        self.dynamic = None
+
+    def _split_input(self, cubes: list):
+        """
+        Extracts convective and dynamic components from the list as objects on the class
+        and ensures units are m s-1
+        """
+        if not isinstance(cubes, iris.cube.CubeList):
+            cubes = iris.cube.CubeList(cubes)
+        self.convective = self._get_cube(cubes, "convective_precipitation_rate")
+        self.dynamic = self._get_cube(cubes, "dynamic_precipitation_rate")
+        if self.convective is None:
+            raise RuntimeError("self.convective is still undefined!!")
+
+    def _get_cube(self, cubes: iris.cube.CubeList, name: str):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5973665)